### DISCUSSION TITLE
Guide to installing native PHP extensions?

### DISCUSSION THREAD

**Discussion by webspec-zebb** [beyondcode/herd-community#151] [Q&A]
My understanding is pecl extensions like xdebug are not included with Herd (which is fine), but I surprised to see the XSL extension was not. When I searched homebrew for `php81-xsl` I could not find a quick way run herd's php binary with XSL.  Has anyone ran into this yet? How can I setup Herd so I can install packages that depend on native PHP extensions?

**Comment by MACscr** [comment]
Im in the same boat when it comes to easily installing extensions and the lack of docs much help community is making it hard to not want to switch back to valet, where there is already a bunch of resources available

**Reply by mpociot** [reply]
I understand the confusion about the different types of extensions. 
Built-in PHP extensions such as XSL, IMAP, and others need to be enabled at compile time. 
Because PHP gets compiled statically, all of the extension dependencies need to be bundled with PHP itself. 
This is done via https://github.com/crazywhalecc/static-php-cli which does not support all extensions (yet)

I just noticed that XSL support is available now though, so I'll publish a set of fresh PHP binaries with XSL included 👍
